### PR TITLE
Fixed issues with distributed training and added example for Imagenet with inception V3

### PR DESCRIPTION
**Pull request by jorgemf**
There was a race condition when running the distributed examples using the last worker as an evaluator server. 

An example for distributed ImageNet with InceptionV3 training is added but still needs a bit more of testing to fine tune the parameters.

### CODE REVIEW

File: example/tutorial_imagenet_inceptionV3_distributed.py [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+from tensorflow.contrib.slim.python.slim.nets.inception_v3 import (inception_v3,
+                                                                   inception_v3_arg_scope)
+
+from tensorflow.python.framework.errors_impl import OutOfRangeError
+from tensorflow.python.training import session_run_hook
+from tensorflow.python.training.basic_session_run_hooks import StopAtStepHook
+from tensorflow.python.training.monitored_session import SingularMonitoredSession
+
+########## VARIABLES ##########
+
```
[the remark this answers is not in the document]
Comment (jorgemf) [reply example/tutorial_imagenet_inceptionV3_distributed.py]: @luomai I have removed some empty lines but I am not sure what empty lines you meant exactly

**Comment by lgarithm** [comment]
how to prepare the dataset?

I got the following error when running in a clean repository
```
$ python example/tutorial_imagenet_inceptionV3_distributed.py 
2018-01-10 14:39:53,067 Batch size: 32
2018-01-10 14:39:53,067 Epochs: 100
Traceback (most recent call last):
  File "example/tutorial_imagenet_inceptionV3_distributed.py", line 429, in <module>
    might_create_training_set()
  File "example/tutorial_imagenet_inceptionV3_distributed.py", line 82, in might_create_training_set
    shuffle=True)
  File "example/tutorial_imagenet_inceptionV3_distributed.py", line 64, in might_create_dataset
    for filename in tf.gfile.Glob(os.path.join(annotations_dir, suffix)):
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/lib/io/file_io.py", line 335, in get_matching_files
    compat.as_bytes(single_filename), status)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/errors_impl.py", line 473, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: ./ILSVRC/Annotations/CLS-LOC/train; No such file or directory
```

**Comment by jorgemf** [comment]
@lgarithm https://github.com/jorgemf/tensorlayer/blob/3c027c7c7fece77efba9f1c24944ba2b38f20522/example/tutorial_imagenet_inceptionV3_distributed.py#L4-L7

You need to accept the terms and conditions before downloading the dataset. I can add a message to handle that error.

**Comment by lgarithm** [comment]
@luomai I'm OK with the current PR.
@jorgemf handling that error would be nice, but if it is tricky to handle, you can just add more comments in the beginning of file, saying that data must be properly prepared in order to run it.

**Comment by jorgemf** [comment]
@lgarithm I am adding a message now, let me test it and upload the change.

**Pull request merged**